app/src/main/java/com/kylehanish/todo/fragments/TodoDialogFragment.java

```diff
@@ -0,0 +1,148 @@
+package com.kylehanish.todo.fragments;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+
+import com.kylehanish.todo.R;
+import com.kylehanish.todo.classes.TodoItem;
+import com.kylehanish.todo.interfaces.iTodoItemChangeListener;
+
+import java.time.LocalDateTime;
+import java.util.Calendar;
+import java.util.Date;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+
+/**
+ * Created by Kyle Hanish on 4/29/18.
+ */
+
+public class TodoDialogFragment extends DialogFragment {
+
+
+    @BindView(R.id.inputDescription)
+    EditText inputDescription;
+
+
+    public static final String TAG = TodoDialogFragment.class.getSimpleName();
+    public static final String BUNDLE_TODO_ITEM = "todo_item";
+    public static final String BUNDLE_DESRIPTION_VALUE = "et_description_value";
+
+    private Unbinder mUnbinder;
+    private iTodoItemChangeListener mListener;
+
+    private TodoItem mCurrentItem;
+    private String mCurrentDecriptionValue;
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+
+        setValuesFromBundle(getArguments(),savedInstanceState);
+
+        String titleString = mCurrentItem.getID() > 0 ? getString(R.string.edit_item) : getString(R.string.add_new_item);
+
+        builder.setTitle(titleString);
+         builder.setPositiveButton(getString(R.string.save), new DialogInterface.OnClickListener() {
+             @Override
+             public void onClick(DialogInterface dialog, int which) {
+                 if(mCurrentItem.getID() == 0){
+                     mCurrentItem.setCreatedOn(Calendar.getInstance().getTime());
+                 }
+                 mCurrentItem.setDescription(inputDescription.getText().toString());
+                mCurrentItem.setCreatedOn(Calendar.getInstance().getTime());
+
+
+                 mListener.SaveTodoItem(mCurrentItem,mCurrentItem.getID() > 0);
```

Comment:
Before saving could do a params check to make sure their item isn't an empty string.